src/openforms/registrations/contrib/objects_api/submission_registration.py

```diff
@@ -228,6 +241,51 @@ def get_update_payment_status_data(
         return render_to_json(options["payment_status_update_json"], context)
 
 
+class ObjectsAPIV2Handler(ObjectsAPIRegistrationHandler[RegistrationOptionsV2]):
+
+    @override
+    def get_object_data(
+        self, submission: Submission, options: RegistrationOptionsV2
+    ) -> dict[str, Any]:
+        submission_variables = get_variables_for_context(submission)
```

Comment:
as remarked in the office - this is intended to be used with the Django template engine which is explicitly what we don't want to do here.

A more correct, low-level and consistent approach is to use:

```py
submission.load_submission_value_variables_state()
```
see also `openforms.submissions.logic.datastructures.DataContainer` which works with this, in particular there's the `to_python` method on variables that may need to be used to normalize data into the python domain.